gwcs/api.py

```diff
@@ -289,3 +289,23 @@ def world_to_array_index(self, *world_objects):
         """
         result = self.invert(*world_objects, with_units=True)[::-1]
         return tuple([utils._toindex(r) for r in result])
+
+    @property
+    def pixel_axis_names(self):
+        """
+        An iterable of strings describing the name for each pixel axis.
+        """
+        if self.input_frame:
+            return list(self.input_frame.axes_names)
+        else:
+            return [''] * self.pixel_n_dim
```

Comment:
`gwcs` allows `input_frame` and `output_frame` to be strings. In this case the API will fail because `WCS.input_frame` is not an object and has so none of the attributes exist. 
Initially this was put in gwcs to make it easy to test , now I wonder if it should be removed.

Reply:
What do you suggest as a mitigation here? Test for `isinstance(str)`?

Reply:
#263 fixed this. Noe `WCS.input_frame` returns either a `CordinateFrame` object or None (including the case when it is a string). If None, `WCS.pixel_n_dim` an `WCS.world_n_dim` return 
`n_inputs` and `n_outputs` of the forward transform. So please change the two methods to check for None on lines 326 and 336. And squash commits.